### PR TITLE
functionality to check gui password expiry and events generation

### PR DESCRIPTION
**Pull request by badri-pathak**
## Pull request checklist

Zenhub Issue: [https://github.ibm.com/IBMSpectrumScale/csi-tracker/issues/38](https://github.ibm.com/IBMSpectrumScale/csi-tracker/issues/38)

```release-notes

```

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature Enhancement
- [ ] Test Automation 
- [ ] Code Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Community Operator listing 
- [ ] Other (please describe): 

## What is the current behavior?

- currently we are not validating whether the GUI password has been expired or still-active. Only validation is happening at the csi setup time.

## What is the new behavior?

- The new behavior will add a reconcile functionality in every 24hours to verify gui password expiry. If there is an issue in the credentials details/expired, events will be generated in the CSO and CSO status will remain same.

## How risky is this change?
- [ ] Small, isolated change
- [x] Medium, requires regression testing
- [ ] Large, requires functional and regression testing

### CODE REVIEW

**Comment by badri-pathak** [comment]
Image : quay.io/badri_pathak/ibm-spectrum-scale-csi-operator:events_gui_pass_v6

File: operator/controllers/csiscaleoperator_controller.go [hunk 1]
```diff
@@ -2381,3 +2398,23 @@ func setDefaultDriverEnvValues(data map[string]string) {
...
+			if err != nil && isUnauthorized(err) {
+				expiredGui = append(expiredGui, cls.Id)
+			}
+		}
+	}
+	return expiredGui
+}
+
+func isUnauthorized(err error) bool {
+	return strings.Contains(err.Error(), "Unauthorized")
```
Comment (amdabhad) [inline operator/controllers/csiscaleoperator_controller.go:2419]: Existing const ErrorUnauthorized="401: Unauthorized" does not work?

**Comment by badri-pathak** [comment]
New image: quay.io/badri_pathak/ibm-spectrum-scale-csi-operator:events_gui_pass_v8

File: operator/controllers/csiscaleoperator_controller.go (hunk 1 above)
Comment (badri-pathak) [reply operator/controllers/csiscaleoperator_controller.go]: Have used constant. would resolve this issue.

**Comment by badri-pathak** [comment]
This is the sample output in the CSO events.

> Warning  AuthError          25s (x3 over 4m40s)  CSIScaleOperator  Either the username/password is incorrect or the password has been expired for the Scale GUI clusterIds: [1766310189841330368 4535625177255832744]

**Review by Jainbrt — APPROVED** [review]

**Comment by badri-pathak** [comment]
Issue will be fixed in this PR [https://github.com/IBM/ibm-spectrum-scale-csi/issues/1003](https://github.com/IBM/ibm-spectrum-scale-csi/issues/1003)

**Comment by badri-pathak** [comment]
New image to verify : quay.io/badri_pathak/ibm-spectrum-scale-csi-operator:events_gui_pass_v9

**Review by Jainbrt — APPROVED** [review]

**Pull request merged**